lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -74,202 +106,71 @@ func (s *subscriberdbServicer) ListSubscribers(ctx context.Context, req *lte_pro
-	digest, noUpdates := s.getDigestInfo(req, networkID, apnsByName, apnResourcesByAPN)
-	if noUpdates {
-		subProtos = []*lte_protos.SubscriberData{}
+	digest, err := s.getDigest(networkID)
+	if err != nil {
+		// The error is swallowed to not affect the main functionality
+		glog.Errorf("Generating digest for network %s failed: %+v", networkID, err)
+		digest = ""
 	}
 
 	listRes := &lte_protos.ListSubscribersResponse{
 		Subscribers:   subProtos,
 		NextPageToken: nextToken,
-		Digest:        digest,
-		NoUpdates:     noUpdates,
+		Digest:        &lte_protos.SubscribersDigest{Md5Base64Digest: digest},
 	}
-
 	return listRes, nil
 }
 
 // getDigestInfo returns the correctly formatted SubscribersDigest and NoUpdates
-// values in response to a list subscribers request.
+// values according to the client digest.
 func (s *subscriberdbServicer) getDigestInfo(
-	req *lte_protos.ListSubscribersRequest,
+	clientDigest *lte_protos.SubscribersDigest,
 	networkID string,
-	apnsByName map[string]*lte_models.ApnConfiguration,
-	apnResourcesByAPN lte_models.ApnResources,
 ) (*lte_protos.SubscribersDigest, bool) {
-	digestProto, noUpdates := req.Digest, false
-	// This functionality is currently placed behind a feature flag
-	if !s.flatDigestEnabled || req.PageToken != "" {
-		return digestProto, noUpdates
+	// The flat digest functionality is currently placed behind a feature flag
+	if !s.flatDigestEnabled {
+		return &lte_protos.SubscribersDigest{Md5Base64Digest: ""}, false
 	}
 
-	digest, err := getDigest(networkID, apnsByName, apnResourcesByAPN)
+	digest, err := s.getDigest(networkID)
 	// If digest generation fails, the error is swallowed to not affect the main functionality
 	if err != nil {
-		glog.Errorf("Generating digest for subscribers in network of gateway %s failed", networkID)
-	} else {
-		digestProto = &lte_protos.SubscribersDigest{Md5Base64Digest: digest}
-		noUpdates = req.Digest.Md5Base64Digest == digest
+		glog.Errorf("Generating digest for network %s failed: %+v", networkID, err)
+		return &lte_protos.SubscribersDigest{Md5Base64Digest: ""}, false
 	}
-	return digestProto, noUpdates
-}
 
-// getDigest returns a deterministic digest of all subscribers in the network.
-// Digest is generated by loading all subscribers in the network.
-func getDigest(
-	networkID string,
-	apnsByName map[string]*lte_models.ApnConfiguration,
-	apnResourcesByAPN lte_models.ApnResources,
-) (string, error) {
-	subProtosById := map[string]proto.Message{}
-	curToken := ""
-	foundEmptyToken := false
-	for !foundEmptyToken {
-		subProtos, nextToken, err := loadSubProtosPage(0, curToken, networkID, apnsByName, apnResourcesByAPN)
-		if err != nil {
-			return "", err
-		}
-		for _, subProto := range subProtos {
-			subProtosById[subProto.Sid.Id] = subProto
-		}
-		foundEmptyToken = nextToken == ""
-		curToken = nextToken
-	}
-	return mproto.HashManyDeterministic(subProtosById)
+	noUpdates := digest != "" && clientDigest != nil && clientDigest.Md5Base64Digest == digest
```

Comment:
This is a case where you can use a guarded getter -- to coalesce the second and third conditions